README.md

```diff
@@ -287,6 +288,28 @@ DATABASE_URL="clickhouse://username:password@127.0.0.1:9000/database_name?on_clu
 
 [See other supported connection options](https://github.com/ClickHouse/clickhouse-go#dsn).
 
+#### BigQuery
+Follow the following format for `DATABASE_URL` when connecting to actual BigQuery in GCP:
+
+```
+bigquery://projectid/location/dataset?disable_auth=true
+```
+`projectid` (mandatory) - Project ID
+
+`dataset` (mandatory) - Dataset Name within the Project
+
+`location` (optional) - Where Dataset is created
+
+`disable_auth` (optional) - Pass `true` to skip Authentication, use only for testing or connecting to emulator.
+
+*NOTE: Follow [this doc](https://cloud.google.com/docs/authentication/provide-credentials-adc) on how to set `GOOGLE_APPLICATION_CREDENTIALS` environment variable for proper Authentication*
+
+Follow the following format if trying to connect to a custom endpoint e.g. [BigQuery Emulator](https://github.com/goccy/bigquery-emulator)
+
+```
+bigquery://host:port/projectid/location/dataset
+```
```

Comment:
Should be:

```
bigquery://host:port/projectid/dataset
```

Reply:
Since the underlying driver supports `location` should dbmate support location path as well?

Reply:
added success test for success `Ping()` and `MigrationsTableExists()`

Reply:
the emulator does not support INFORMATION tables so not sure how I can follow the pattern like other driver tests for bq

Reply:
> added success test for success `Ping()` and `MigrationsTableExists()`

Okay, now, temporarily change `BIGQUERY_TEST_URL` to `bigquery://bigquery:9050/test/location/dbmate_test?disable_auth=true` in your `docker-compose.yml` (no need to check this in) and run the tests.

Reply:
made the change and ran the tests. all tests passed. recorder: https://asciinema.org/a/T6K53MgHb6I89Lm3tFz58tOqN

Reply:
> the emulator does not support INFORMATION tables so not sure how I can follow the pattern like other driver tests for bq

We can implement special tests that will only run against an actual Google BigQuery database URL conditionally, but we have to be very careful as the tests create and drop the dataset, which if someone runs the tests against an actual dataset in use, could be disasterous.

I'm talking about something like this:

```go
func TestGoogleBigQueryDumpSchema(t *testing.T) {
	t.Run("default migrations table", func(t *testing.T) {
		drv := testBigQueryDriver(t)

		if drv.databaseURL.Host != "" {
			t.Skipf("skipping test, BIGQUERY_TEST_URL pointing at %s and not at Google", drv.databaseURL.Host)
		}

		// test code continues here
	})
}
```

Reply:
I’m not sure I understand how these specific test cases would work in CI? If the test url connects to an actual BQ instance then it will pass/fail otherwise pass?

Reply:
@dossy i didn’t understand your comment about dbmate accepting host or host:port. That is how it’s working at

Reply:
> I’m not sure I understand how these specific test cases would work in CI? If the test url connects to an actual BQ instance then it will pass/fail otherwise skip?

In CI, only test cases that can run against the emulator will be run; the tests that require the real Google BigQuery service should be skipped.

This way, we can implement tests that require the real Google BigQuery service to ensure that we're testing all the necessary functionality, but not put undue burden on anyone who wants to run the tests without having to set up a real GCP account, etc.

It makes accepting contribution for those drivers a little more difficult, but the alternatives are either not supporting drivers that have this requirement which hurts everyone, or not testing functionality that can only be tested against the real thing, which makes it hard to trust that dbmate will do the right thing.

> It might also be feasible to run the special cases against a free BQ instance? https://cloud.google.com/bigquery?hl=en#pricing

It might be possible to set up a free-tier GCP account with properly defined IAM permissions on a Service Account that could be used for testing in CI.  @amacneil, would you be willing to add a secret to the GitHub repo with a test GCP account's credentials JSON for use in CI test runs?

Reply:
> i didn’t understand your comment about dbmate accepting host or host:port. That is how it’s working atm

I think it was a mistake for me to suggest that we support encoding the host and port in the connection string URL directly.

The [BigQuery Service Endpoint](https://cloud.google.com/bigquery/docs/reference/rest#service-endpoint) is `https://bigquery.googleapis.com`, so technically one could reasonably use the following connection string URL:

`bigquery://project/location/dataset?endpoint=https%3A%2F%2Fbigquery.googleapis.com`

The key issue here is being able to specify the correct scheme to use for the endpoint (e.g., `http` vs `https`), which if we try to infer the endpoint URL from the hostname and port portions of the connection string URL, we would not be able to infer the correct scheme for the endpoint.  We could try to assume that 443 means `https`, but does, say, port `1234` require `http` or `https`?  Can't know for sure, and there's no obvious way to encode it into the connection string.

In other words: now that we can get the project ID, location, and dataset from the underlying driver, we should probably just pass through the connection string URL as-is through to the underlying `go-gorm/bigquery` driver, let it do its connection string URL parsing, including encoding the endpoint as a query parameter, and read the values from the underlying `config` structure.

I'll push up a few commits that will hopefully clarify all of this.